src/Polysemy/NonDet.hs

```diff
@@ -106,18 +103,24 @@ instance Monad (NonDetC m) where
     a (\ a' -> unNonDetC (f a') cons)
   {-# INLINE (>>=) #-}
 
+instance MonadTrans NonDetC where
+  lift m = NonDetC $ \c b -> m >>= (`c` b)
+
+instance MonadTransControl NonDetC where
+  type StT NonDetC = []
+
+  hoistT n nd = NonDetC $ \c b ->
+    join $ n $ unNonDetC nd (\a r -> return $ c a (join (n r))) (return b)
```

Comment:
Note: `NonDetC` is not a well-behaved `MonadTransControl` (due to #246), but this definition of `hoistT` _is_ well-behaved (morally, `hoistT id == id`). This gives rise to an inconsistency, where:
```haskell
controlT (\lower -> n (lower tm)) /= hoistT n tm
```